src/Modules/KnowledgeComponents/Tutor.KnowledgeComponents.Infrastructure/KnowledgeComponentsStartup.cs

```diff
@@ -91,8 +91,8 @@ private static void SetupInfrastructure(IServiceCollection services)
 
         services.AddScoped<IKnowledgeMasteryRepository, KnowledgeMasteryDatabaseRepository>();
 
-        services.AddScoped<IEventStore, PostgresStore>();
-        services.AddSingleton<IEventSerializer>(new DefaultEventSerializer(EventSerializationConfiguration.EventRelatedTypes));
+        services.AddScoped<IKnowledgeComponentEventStore, PostgresStore>();
```

Comment:
It seems that both `IEventStore` and `IEventSerializer` can remain in `BuildingBlocks.Infrastructure` if we make the interfaces parametrized.

Look to the generic repositories:
- Generic database repo https://github.com/Clean-CaDET/tutor/blob/master/src/BuildingBlocks/Tutor.BuildingBlocks.Infrastructure/Database/CrudDatabaseRepository.cs
- Startup configuration https://github.com/Clean-CaDET/tutor/blob/master/src/Modules/Courses/Tutor.Courses.Infrastructure/CoursesStartup.cs#L56